Makefrag

```diff
@@ -26,17 +27,21 @@ OBJDUMP := $(RISCV_PREFIX)objdump
 OBJCOPY := $(RISCV_PREFIX)objcopy
 EMU ?= $(FLEXPRET_ROOT_DIR)/emulator/fp-emu # Verilator C++ emulator
 
-include $(FLEXPRET_ROOT_DIR)/defines.mk
-
-CFLAGS = -g -static -O0 -march=rv32i -mabi=ilp32 -nostartfiles --specs=nosys.specs $(APP_DEFS) $(DEFINES)
-INCS = $(LIB_INCS) $(APP_INCS)
-LFLAGS = -T $(LINKER_SCRIPT) -L $(FP_LIB_DIR)/linker -Xlinker -Map=$(NAME).map 
+CFLAGS += -g -static -O0 -march=rv32i -mabi=ilp32 -nostartfiles --specs=nosys.specs -Werror=stack-usage=$(STACKSIZE) $(APP_DEFS) $(DEFINES)
+INCS += $(LIB_INCS) $(APP_INCS)
+LFLAGS += -T $(LINKER_SCRIPT) -L $(FP_LIB_DIR)/linker -Xlinker -Map=$(NAME).map 
 
+ifeq ($(WANT_DEBUG),true)
+CFLAGS += -fstack-protector-all
+endif
```

Comment:
`-f-stack-protector-all` modifies all function calls to be able to detect stack error in run-time. Doing this to all functions might be a bit much, so it is wrapped in the WANT_DEBUG if statement. 

In the normal case, `-f-stack-protector` should be used, which does this to just a select few functions. That is not the case in the code as of now though, so I'll update it.